src/query/functions-v2/src/aggregates/aggregate_min_max_any.rs

```diff
@@ -0,0 +1,287 @@
+    ) -> Result<()> {
+        let column = T::try_downcast_column(&columns[0]).unwrap();
+        let column_iter = T::iter_column(&column);
+        column_iter.zip(places.iter()).for_each(|(v, place)| {
+            let addr = place.next(offset);
+            let state = addr.get::<State>();
+            state.add(v.clone())
+        });
+        Ok(())
+    }
+    fn accumulate_row(&self, place: StateAddr, columns: &[Column], row: usize) -> Result<()> {
+        let column = T::try_downcast_column(&columns[0]).unwrap();
+        let v = T::index_column(&column, row);
+        if let Some(v) = v {
+            let state = place.get::<State>();
+            state.add(v)
+        }
+        Ok(())
+    }
+
+    fn serialize(&self, place: StateAddr, writer: &mut BytesMut) -> Result<()> {
+        let state = place.get::<State>();
+        state.serialize(writer)
+    }
+
+    fn deserialize(&self, place: StateAddr, reader: &mut &[u8]) -> Result<()> {
+        let state = place.get::<State>();
+        state.deserialize(reader)
+    }
+
+    fn merge(&self, place: StateAddr, rhs: StateAddr) -> Result<()> {
+        let rhs = rhs.get::<State>();
+        let state = place.get::<State>();
+        state.merge(rhs)
+    }
+
+    fn merge_result(&self, place: StateAddr, builder: &mut ColumnBuilder) -> Result<()> {
+        let state = place.get::<State>();
+        state.merge_result(builder)
+    }
+
+    fn need_manual_drop_state(&self) -> bool {
+        self.need_drop
+    }
+
+    unsafe fn drop_state(&self, place: StateAddr) {
+        let state = place.get::<State>();
+        std::ptr::drop_in_place(state);
+    }
+}
+
+impl<T, C, State> fmt::Display for AggregateMinMaxAnyFunction<T, C, State> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+impl<T, C, State> AggregateMinMaxAnyFunction<T, C, State>
+where
+    T: ValueType + Send + Sync,
+    C: ChangeIf<T> + Default,
+    State: ScalarStateFunc<T>,
+{
+    pub fn try_create(
+        display_name: &str,
+        return_type: DataType,
+        need_drop: bool,
+    ) -> Result<Arc<dyn AggregateFunction>> {
+        let func = AggregateMinMaxAnyFunction::<T, C, State> {
+            display_name: display_name.to_string(),
+            data_type: return_type,
+            need_drop,
+            _t: PhantomData,
+            _c: PhantomData,
+            _state: PhantomData,
+        };
+        Ok(Arc::new(func))
+    }
+}
+
+pub fn try_create_aggregate_min_max_any_function<const TYPE: u8>(
+    display_name: &str,
+    _params: Vec<Scalar>,
+    argument_types: Vec<DataType>,
+) -> Result<Arc<dyn AggregateFunction>> {
+    assert_unary_arguments(display_name, argument_types.len())?;
+    let data_type = argument_types[0].clone();
+    let need_drop = need_manual_drop_state(&data_type);
+
+    with_basic_no_number_mapped_type!(|T| match data_type {
+        DataType::T => {
+            if TYPE == TYPE_MIN {
```

Comment:
TYPE can migrate into other traits.

```
trait TYPE {
    type CmpType: ChangeIf ...
}
```

Reply:
It's not easy to migrate this, because `ChangeIf` is bounded with `T: ValueType`.